firestore/src/main/firebase_metadata_provider_desktop.cc

```diff
@@ -16,28 +16,27 @@
 
 #include "firestore/src/main/firebase_metadata_provider_desktop.h"
 
-#include "app/src/heartbeat_info_desktop.h"
 #include "app/src/include/firebase/app.h"
 
 namespace firebase {
 namespace firestore {
 
+const char* kHeartbeatCodeGlobal = "2";
+
 FirebaseMetadataProviderCpp::FirebaseMetadataProviderCpp(const App& app)
-    : gmp_app_id_(app.options().app_id()) {}
+    : heartbeat_controller_(Move(app.GetHeartbeatController())),
```

Comment:
Although it may be technically safe, this `Move` here makes me nervous. Can it be safely removed?

Reply:
This move was just meant to avoid copying a shared pointer. It could be removed, and a copy of the shared pointer would be created.

Reply:
Ohh I see. I missed that `app.GetHeartbeatController()` was returning a shared_ptr. Please ignore this comment.